### PR TITLE
use shared client for throttling per host

### LINKED ISSUE #45: Throttle # of connections to api
When Ajour performs a refresh or update all operation, a large number of simultaneous asynchronous calls are being made to the API (curse, tukui, wowinterface). I suggest a setting that caps/throttles the number of connections.

I do not think this value needs to be exposed in the config file / settings view as most users will probably see it as a way to "optimize" their performance and not consider the API. But if we do expose it, it should only allow a range of sensible values.

For reference, a typical value used by browsers is 6 per server.

See Firefox about.config setting: network.http.max-persistent-connections-per-server

### PR DESCRIPTION
**Pull request by tarkah**
resolves #45 

Haven't tested this as I don't have any WoW installed / addons on my machine. It should limit 6 connections per API host by using a shared client across all requests and specifying `max_connections_per_host(6)`

### CODE REVIEW

File: src/gui/mod.rs [hunk 1]
```diff
@@ -80,6 +86,13 @@ impl Default for Ajour {
             config: Config::default(),
             expanded_addon: None,
             is_showing_settings: false,
+            shared_client: Arc::new(
+                HttpClient::builder()
+                    .redirect_policy(RedirectPolicy::Follow)
+                    .max_connections_per_host(6)
+                    .build()
+                    .unwrap(),
```
Comment (tarkah) [inline src/gui/mod.rs:94]: Program can panic here, not sure how you want to handle this inside `<Ajour as Application>::new`

**Comment by casperstorm** [comment]
Hi @tarkah.
I've tested this and it works great. This is a clever way of solving this, and is much nicer against the different API's.
Thank you so much for contributing.

File: src/gui/mod.rs (hunk 1 above)
Comment (casperstorm) [reply src/gui/mod.rs:94]: Good question. For now I think its okay to continue and tackle this task while doing improvements later.

**Comment by tarkah** [comment]
Great, glad it worked out! I stumbled across this project today and it's awesome. Wish this existed back when I was still playing classic.

**Comment by casperstorm** [comment]
> Great, glad it worked out! I stumbled across this project today and it's awesome. Wish this existed back when I was still playing classic.

I am so glad you like it, and even more glad you are helping contribute!

**Pull request merged**